### PR TITLE
Added make file to hide scripts

### LINKED ISSUE #27: Add make target to build sample
After cloning the repository, there should be a one line command to build the sample application(s)z

Context from PG/Amaury: 
- Takuro will be adding 3 applications to the ccf-app-samples repository (banking app, logging app, contact discovery app)
- Build make in an extensible way for future sample applications being added
- We want to make this repo easier to use from DX example; thus, introducing make files to build + test code pointing to one app. 
- Only focus on JS Samples (as C++ will be phased out)
- Keep in mind: the sample applications will ultimately be sample containerized applications with many different languages (make + docker? -- keep this in mind in make design decisions)

### LINKED ISSUE #29: CI should run test target
When the CI is triggered it should build the devcontainer and use that to run the tests - this way we know that the devcontainer always works

Context from PG/Amaury: 
- Takuro will be adding 3 applications to the ccf-app-samples repository (banking app, logging app, contact discovery app)
- Build make in an extensible way for future sample applications being added
- We want to make this repo easier to use from DX example; thus, introducing make files to build + test code pointing to one app. 
- Only focus on JS Samples (as C++ will be phased out)
- Keep in mind: the sample applications will ultimately be sample containerized applications with many different languages (make + docker? -- keep this in mind in make design decisions)

### LINKED ISSUE #30: Update Readme.md based on "new" developer experience with make files
Update the documents so that they "flow" better as a new developer to the repository

### PR DESCRIPTION
**Pull request by ross-p-smith**
We have 2 makefiles. The Makefile in the root is mainly used to lint and clean. If you do try and `make build` from the root, you will receive a message to change directory.

There will be a Makefile per sample that is responsible for building/testing/demoing and any special starting of the network that is required.

I have also added a separate job in the CI build that actually uses the devcontainer to build and run the tests; this way we can always proof that the devcontainer itself works. It also simplifies the build as we can just call `make test` in the CI.

Relates to #27
Relates to #28
Fixes #29
Fixes #30

### CODE REVIEW

**Comment by ross-p-smith** [comment]
I 'think' this may be blocked by #25

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,43 @@
+SHELL := /bin/bash
+
+.PHONY: help prepare-demo
+.DEFAULT_GOAL := help
+
+help: ## 💬 This help message :)
+	@grep -E '[a-zA-Z_-]+:.*?## .*$$' $(firstword $(MAKEFILE_LIST)) | awk 'BEGIN {FS = ":.*?## "}; {printf "\033[36m%-20s\033[0m %s\n", $$1, $$2}'
```
Comment (takuro-sato) [inline Makefile:7]: That's a nice idea!

File: Makefile [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+lint: ## 🔍 Lint the code base (but don't fix)
+	@echo -e "\e[34m$@\e[0m" || true
+	@./lint.sh
+
+lint-fix: ## 🌟 Lint and fix the code base
+	@echo -e "\e[34m$@\e[0m" || true
+	@./lint.sh -f
+
+build: ## 🔨 Build the Banking Application
```
Comment (takuro-sato) [inline Makefile:17]: What do you think is the best way to make changes when we add the second sample?

I can think of the following directory structure for example:
```
.
|-- Makefile # Common commands like `lint`. Also, we can add commands to do something for all app like `test-all`.
|-- banking-app
|    |-- Makefile # `test`, `prepare-demo`, etc for banking-app
|--second-app
    |--Makefile # `test`, `prepare-demo`, etc for second-app

```

Or just we can edit this Makefile and provide command like `make test banking-app` ?

File: Makefile [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+	@echo -e "\e[34m$@\e[0m" || true
+	@cd banking-app && ./test.sh
+
+start-host: build ## 🏃 Start the CCF Sandbox
+	@echo -e "\e[34m$@\e[0m" || true
+	@cd banking-app && /opt/ccf/bin/sandbox.sh --js-app-bundle ./dist/ --initial-member-count 3 --initial-user-count 0
+
+prepare-demo: ## 🚀 Prepare the banking application demo
+	@echo -e "\e[34m$@\e[0m" || true
+	@cd banking-app && ./prepare_demo.sh
```
Comment (takuro-sato) [inline Makefile:31]: I think this make file is better than `prepare_demo.sh`. So how about just removing `prepare_demo.sh` and `prepare-demo`?

File: banking-app/README.md [hunk 4]
```diff
@@ -125,20 +110,33 @@ sequenceDiagram
 ### How to run the demo
 
 ```bash
-cd banking-app
-./prepare_demo.sh
+make prepare-demo
```
Comment (takuro-sato) [inline banking-app/README.md:113]:
```suggestion
make start-host
```

Only we remove `prepare-demo`

File: Makefile (hunk 2 above)
Comment (ross-p-smith) [reply Makefile]: I have added another makefile in the banking-app folder. If you run `make build` from the root you will get the message "Please change directory to the sample you wish to build." When we get a second application I think we can improve this, but for now, you have to be inside the banking-app folder to run `make build` etc.

File: Makefile (hunk 3 above)
Comment (ross-p-smith) [reply Makefile]: I have removed prepare_demo and moved the copy of "constitution" file into the makefile.

We now `make start-host` and `make demo`

File: banking-app/README.md (hunk 4 above)
Comment (ross-p-smith) [reply banking-app/README.md]: Done!

File: Makefile [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+
+demo: ## 🎬 Demo an Application
+	@echo -e "\e[34mPlease change directory to the sample you wish to demo.\e[0m" || true
+
+clean: ## 🧹 Clean the working folders created during build/demo
+	@rm -rf .venv_ccf_sandbox
+	@rm -rf workspace
+	@rm -rf banking-app/.venv_ccf_sandbox
+	@rm -rf banking-app/dist
+	@rm -rf banking-app/workspace
```
Comment (julioalex-rezende) [inline Makefile:34]: Should the general make clean target clean the apps as well?
If we add extra apps to this repo in the future (will/can this happen?). we will need to constantly update this. 

is this the real intention?

Comment (ross-p-smith) [reply Makefile]: I've removed the clean of the banking-app files from the root.

**Pull request merged**